tests/base/src/test/java/org/wildfly/security/auth/client/CredentialStoreSaslAuthenticationTest.java

```diff
@@ -111,4 +112,29 @@ public void testSuccessfulSaslAuthenticationWithCredentialStore() throws Excepti
 
     }
 
+    @Test
+    public void testUnsuccessfulSaslAuthenticationWithCredentialStore() throws Exception {
+        SaslServer server = new SaslServerBuilder(PlainSaslServerFactory.class, PLAIN)
+                .setProviderSupplier(() -> providers)
+                .setUserName(USERNAME)
+                .setPassword((USERNAME + PASSWORD).toCharArray())
+                .build();
+
+        // Create SASL client from XML configuration file
+        AuthenticationContext context = AuthenticationContext.getContextManager().get();
+
+        AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
+        SaslClient client = contextConfigurationClient.createSaslClient(new URI(CREDENTIAL_CONFIG_FILE), context.authRules.getConfiguration(), Arrays.asList(new String[]{PLAIN}));
```

Comment:
It would be good to add a comment here to indicate the password being specified by the client is different from what the server is expecting, e.g., something like

// XML configuration file specifies PASSWORD instead of USERNAME + PASSWORD